src/System.IO.Packaging/src/System/IO/Packaging/ContentType.cs

```diff
@@ -0,0 +1,727 @@
+        /// </summary>
+        /// <param name="contentType">Content type to be compared with</param>
+        /// <returns></returns>
+        internal bool AreTypeAndSubTypeEqual(ContentType contentType)
+        {
+            return AreTypeAndSubTypeEqual(contentType, false);
+        }
+
+        /// <summary>
+        /// This method does a weak comparison of the content types. We only compare the
+        /// type and subType values in an ASCII case-insensitive manner.
+        /// Parameter and value pairs are not used for the comparison. 
+        /// If you wish to compare the paramters too, then you must get the ParameterValuePairs from
+        /// both the ContentType objects and compare each parameter entry.
+        /// The allowParameterValuePairs parameter is used to indicate whether the 
+        /// comparison is tolerant to parameters being present or no.
+        /// </summary>
+        /// <param name="contentType">Content type to be compared with</param>
+        /// <param name="allowParameterValuePairs">If true, allows the presence of parameter value pairs. 
+        /// If false, parameter/value pairs cannot be present in the content type string.
+        /// In either case, the parameter value pair is not used for the comparison.</param>
+        /// <returns></returns>
+        internal bool AreTypeAndSubTypeEqual(ContentType contentType, bool allowParameterValuePairs)
+        {
+            bool result = false;
+
+            if (contentType != null)
+            {
+                if (!allowParameterValuePairs)
+                {
+                    //Return false if this content type object has parameters
+                    if (_parameterDictionary != null)
+                    {
+                        if (_parameterDictionary.Count > 0)
+                            return false;
+                    }
+
+                    //Return false if the content type object passed in has parameters
+                    Dictionary<string, string>.Enumerator contentTypeEnumerator;
+                    contentTypeEnumerator = contentType.ParameterValuePairs;
+                    contentTypeEnumerator.MoveNext();
+                    if (contentTypeEnumerator.Current.Key != null)
+                        return false;
+                }
+
+                // Perform a case-insensitive comparison on the type/subtype strings.  This is a
+                // safe comparison because the _type and _subType strings have been restricted to
+                // ASCII characters, digits, and a small set of symbols.  This is not a safe comparison
+                // for the broader set of strings that have not been restricted in the same way.
+                result = (String.Compare(_type, contentType.TypeComponent, StringComparison.OrdinalIgnoreCase) == 0 &&
+                          String.Compare(_subType, contentType.SubTypeComponent, StringComparison.OrdinalIgnoreCase) == 0);
```

Comment:
These could use String.Equals instead of String.Compare, right?  I know there are a variety of such cleanup things you're hoping to defer until after the initial PR; this can be on that list (which would ideally be opened as an issue so that we can track it).